django/utils/numberformat.py

```diff
@@ -68,7 +68,8 @@ def format(
             str_number = "{:f}".format(number)
     else:
         str_number = str(number)
-    if str_number[0] == "-":
+    # Test for sign but also database null received as empty string
+    if str_number and str_number[0] == "-":
```

Comment:
Would not it be better to check this at the very beginning and return `None` or `""` immediately?
```python
def format(
    ...
):
    if not str_number:
        return str_number
    use_grouping = (
        use_l10n or (use_l10n is None and settings.USE_L10N)
    ) and settings.USE_THOUSAND_SEPARATOR
    ...
```

Reply:
I wanted to be minimally invasive but you are right. I've made the change now.